File: python/step/count.py
import asyncio
import random
from typing import Tuple, Optional

from lead import Lead


async def count(state: int) -> Tuple[int, Optional[Lead]]:
    print(f"Current count: {state}")
    await asyncio.sleep(1)
    state += 1
    if random.choice([True, False]):
        return state, Lead.COUNT
    else:
        return state, Lead.STOP

File: php/step/count.php
<?php

require_once __DIR__ . '/../lead.php';

function countStep($state) {
    echo "Current count: $state\n";
    sleep(1);
    $state++;

    if (rand(0, 1) < 0.5) {
        return [$state, Lead::COUNT];
    } else {
        return [$state, Lead::STOP];
    }
}

